.github/workflows/maven-release-perform.yml

```diff
@@ -31,15 +35,24 @@ jobs:
       - name: Set up JDK 11
         uses: actions/setup-java@v3
         with:
+          distribution: 'temurin'
           java-version: 11
-      - name: Prepare release settings
+          server-id: sonatype-nexus-staging
+          server-username: SONATYPE_USER
+          server-password: SONATYPE_PWD
+          gpg-private-key: ${{ secrets.GPG_KEY }}
+          gpg-passphrase: GPG_PASSPHRASE
+      - name: Download xmlstarlet
+        run: |
+          sudo apt-get install xmlstarlet
+      - name: Setup SSH
+        uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
```

Comment:
We need to set up an SSH key for the `release:perform` because it uses `-DconnectionUrl=scm:git:git@github.com:checkstyle/checkstyle.git` to pull the repo.

Reply:
We have a manual way of doing this in `copy-site-to-sourceforge.yml`:
https://github.com/checkstyle/checkstyle/blob/ab04b31535028b6afbe45e2c50fe458e5cb43328/.github/workflows/copy-site-to-sourceforge.yml#L25-L35

I don't know if you prefer using `webfactory/ssh-agent@v0.7.0` or doing it manually. Let me know so we can keep it consistent.

Reply:
Usage of this third-party action does not make code even smaller

Reply:
Just a suggestion, why even use SSH? We can skip the SSH setup and just do
```
  -DconnectionUrl=scm:git:https://github.com/stoyanK7/checkstyle.git \
```
instead of what we do right now:
https://github.com/checkstyle/checkstyle/blob/e487327aef4fbb8f36a60a29d487375947d7a488/.ci/maven-release-perform.sh#L31

Reply:
Or do we want to stay with the ssh?

Reply:
we do copy to sourceforge, as it is or original , backup website and most used/indexed web side.
ssh is requied only because it is the only way I know how to upload zip to sourceforge.
it is not an operation of maven deploy.

Reply:
Done.

Reply:
usage of git by ssh , allows to avoid usage of passwords/tokens during push.
I use it on local a lot, I thought everyone is using it.
we do push to sources only in release:prepare not a release:perform so usage of http should be ok.